Microsoft.Toolkit/Helpers/ObjectStorage/ISettingsStorageHelper.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Toolkit.Helpers
+{
+    /// <summary>
+    /// Service interface used to store data using key value pairs.
+    /// </summary>
+    public interface ISettingsStorageHelper
+    {
+        /// <summary>
+        /// Determines whether a setting already exists.
+        /// </summary>
+        /// <param name="key">Key of the setting (that contains object).</param>
+        /// <returns>True if a value exists.</returns>
+        bool KeyExists(string key);
+
+        /// <summary>
+        /// Determines whether a setting already exists in composite.
+        /// </summary>
+        /// <param name="compositeKey">Key of the composite (that contains settings).</param>
+        /// <param name="key">Key of the setting (that contains object).</param>
+        /// <returns>True if a value exists.</returns>
+        bool KeyExists(string compositeKey, string key);
+
+        /// <summary>
+        /// Retrieves a single item by its key.
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved.</typeparam>
+        /// <param name="key">Key of the object.</param>
+        /// <param name="default">Default value of the object.</param>
+        /// <returns>The T object</returns>
+        T Read<T>(string key, T? @default = default(T));
```

Comment:
There's an argument against using escaped keywords as parameter names, as that makes it harder to use with named parameters and can also cause issues with consumers on another languages. We're also not doing this in other bits of the toolkit either I think. Maybe we should consider renaming `@default` to `fallback`, or something?

Reply:
What do you think about just removing the `@default` param altogether? The fallback is a convenience, but not really necessary.

Reply:
I think that makes sense, yeah 😄
Though we should then make the contract here clearer, as in, either document the method as throwing if the key is not found and return a non nullable value, or still marking the return as unconstrained nullable `T?` and document this behavior too, just to ensure that types implementing this will be consistent as well.

Reply:
I do like the default, makes it nice to code from the implementers side easily for lots of settings. That said, extensions could probably handle that pattern vs. the base implementation. Maybe V2 we work on later also implements dictionary or something and we have a general helper for that [like this old discussion](https://codereview.stackexchange.com/questions/110621/dictionary-getvalueordefault).

I know in XAML Studio, I wrote an attribute to define the default and used that on my properties to help with my base serialization helper stuff I used.

Reply:
That's rough! On one side I'd love to leave it in to support your use case, @michael-hawker, but you bring up a great point; That functionality could definitely be accomplished more cleanly in the implementations themselves or with extensions, than enforcement through the interface.

Reply:
I added a new static class, `IObjectStorageHelperExtensions`, with `GetValueOrDefault`, `TryRead`, and `TryDelete` helper functions. Take a look and let me know what you think.